src/vse_sync_pp/analyzers/pmc.py

```diff
@@ -0,0 +1,134 @@
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+    STATE_HOLDOVER_OUT_OF_SPEC3: [STATE_LOCKED, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+}
+
+BASE_CLOCK_CLASS_COUNT = {
+    "count": 0,
+    "transitions": {
+        STATE_FREERUN: 0,
+        STATE_LOCKED: 0,
+        STATE_HOLDOVER_IN_SPEC: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC1: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC2: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC3: 0,
+    },
+}
+
+def is_transition_allowed(current_state, new_state):
+   return new_state in STATE_TRANSITION[current_state]
+
+def get_named_clock_class_result(clock_class_count):
+    named_clock_class_count = {STATE_NAMES[k]:v for (k,v) in clock_class_count.items()}
+    for clock_class in clock_class_count.values():
+        clock_class["transitions"] = {STATE_NAMES[k]:v for (k,v) in clock_class["transitions"].items()}
+    return named_clock_class_count
+
+class ClockStateAnalyzer(Analyzer):
+    """Analyze clock state
+    """
+    id_ = 'phc/gm-settings'
+    parser = id_
+    def __init__(self, config):
+        super().__init__(config)
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        self.state = None
+        self.transition_count = 0
+        self.clock_class_count = {
+            STATE_FREERUN: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_LOCKED: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_IN_SPEC: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_OUT_OF_SPEC1: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_OUT_OF_SPEC2: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+            STATE_HOLDOVER_OUT_OF_SPEC3: copy.deepcopy(BASE_CLOCK_CLASS_COUNT),
+        }
+    def prepare(self, rows):
+        idx = 0
+        try:
+            tstart = rows[0].timestamp
+        except IndexError:
+            pass
+        else:
+            while idx < len(rows):
+                if tstart <= rows[idx].timestamp:
+                    break
+                idx += 1
+        return super().prepare(rows[idx:])
+    def test(self, data):
+        if len(data) == 0:
+            return (False, "no data")
+        
+        for index, row in data.iterrows():
+            clock_class = row['clock_class']
+ 
+            if (self.state is None) and (clock_class in STATE_TRANSITION):
+                self.state = clock_class
+            else:
+                if clock_class != self.state:
+                    self.transition_count += 1
+                
+                if clock_class not in STATE_TRANSITION:
+                    return (False, f"wrong clock class {clock_class}")
```

Comment:
We should review together how we want failure reasons to appear in the test report.